src/YHttpWidgetsActionHandler.cc

```diff
@@ -239,71 +229,143 @@ int YHttpWidgetsActionHandler::do_action(YWidget *widget, const std::string &act
+                input->setKeyboardFocus();
+                input->setValue( value );
+                activate_widget( input );
+            } );
         }
+        else if ( dynamic_cast<YComboBox*>(widget) )
+        {
+            YComboBox* cb = dynamic_cast<YComboBox*>(widget);
+            if( !cb->editable() )
+            {
+                body << "{ \"error\" : \"Combobox '" << cb->label() << "' is not editable \" }" << std::endl;
+                return MHD_HTTP_UNPROCESSABLE_ENTITY;
+            }
+
+            return action_handler<YComboBox>( widget, body, [&] (YComboBox *cb) {
+                yuiMilestone() << "Setting value for YComboBox \"" << cb->label() << '"' << std::endl;
+                cb->setKeyboardFocus();
+                cb->setValue( value );
+                activate_widget( cb );
+            } );
+        }
+
+        std::ostringstream error;
+        error << "Action 'enter_text' is not supported for the selected widget: \"" << widget->widgetClass() << "\"";
+        return handle_error( body, error.str(), MHD_HTTP_NOT_FOUND );
     }
     else if ( action == "select" )
     {
         std::string value;
         if (const char* val = MHD_lookup_connection_value(connection, MHD_GET_ARGUMENT_KIND, "value"))
             value = val;
-
         if ( dynamic_cast<YComboBox*>(widget) )
         {
-            return action_handler<YComboBox>(widget, [&] (YComboBox *cb) {
+            return action_handler<YComboBox>( widget, body, [&] (YComboBox *cb) {
                 yuiMilestone() << "Activating ComboBox \"" << cb->label() << '"' << std::endl;
                 cb->setKeyboardFocus();
-                cb->setValue(value);
+                YItem * item = cb->findItem(value);
+                if ( item )
+                {
+                    yuiMilestone() << "Activating Combobox \"" << cb->label() << '"' << std::endl;
+                    cb->selectItem( item );
+                    activate_widget( cb );
+                }
+                else
+                {
+                    throw YUIException("Item: '" + value + "' cannot be found in the combobox");
+                }
             } );
         }
         else if( dynamic_cast<YTable*>(widget) )
         {
+            int row_id = 0;
+            if ( const char* val = MHD_lookup_connection_value(connection, MHD_GET_ARGUMENT_KIND, "row") )
+                row_id = atoi(val);
+            // Handle case when want select row by row number
+            return action_handler<YTable>( widget, body, [&] (YTable *tb) {
+
+                if( row_id >= tb->itemsCount() ) {
+                    throw YUIException( "Table: '" + tb->label() + "' does NOT contain row #" + std::to_string( row_id ) );
+                }
+                YItem * item = 0;
+                int current_row = 0;
+                for ( YItemConstIterator it = tb->itemsBegin(); it != tb->itemsEnd(); ++it )
```

Comment:
The `YSelectionWidget` parent class implements the [`itemAt(int index)`](https://github.com/libyui/libyui/blob/master/src/YSelectionWidget.h#L359) method. Unfortunately it is protected and cannot be used form outside.

But as it would make sense for this code maybe we should make it public. @shundhammer ?

Reply:
Yes, let's make it public. It would only be consistent. There is also itemsCount() which is also public, so having access to itemAt() would only be logical.

Reply:
Don't forget to adjust the BuildRequires libyui >= ... here when you change that!

Reply:
Alright, I will make relevant change in libyui. Thanks!